st2common/st2common/validators/api/reactor.py

```diff
@@ -65,11 +66,14 @@ def validate_trigger_parameters(trigger_type_ref, parameters):
     if not trigger_type_ref:
         return None
 
-    if trigger_type_ref not in SYSTEM_TRIGGER_TYPES:
-        # Not a system trigger, skip validation for now
+    trigger_type = triggers.get_trigger_type_db(trigger_type_ref)
+    if trigger_type_ref in SYSTEM_TRIGGER_TYPES:
+        parameters_schema = SYSTEM_TRIGGER_TYPES[trigger_type_ref]['parameters_schema']
```

Comment:
> In addition to that, we also need some tests for rule creation with invalid parameters

I had never realized that case, thank you. I'll cover it in the test.

> I think we should have a single config option.

Sure, I'll correct the configuration parameter and make it translate to the `system` section because this flag will affect system wide.

> We also need to document this new flag and behavior in the upgrade notes section of document (clarify how it works and affects the system).

I got it. I'll make another PR to [Upgrade Notes](https://github.com/StackStorm/st2docs/blob/598d7f82e9fba0d8b199a4a5bb15509e1ed051b7/docs/source/upgrade_notes.rst) in st2docs.